src/index.js

```diff
@@ -35,8 +52,8 @@ function update(state=initialState, action) {
 
 // Syncing
 
-function locationToString(location) {
-  return location.pathname + location.search + location.hash;
+function locationsAreEqual(a, b) {
+  return a.path === b.path && deepEqual(a.state, b.state);
```

Comment:
Are you sure we need to `deepEqual` here? I'm having a hard time wrapping my head around this, but probably because I haven't use that state much.

Hm, I have a thought... read my 2nd comment below.